src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -575,12 +575,12 @@ private void WriteMethodPrototype(CodeMethod code, CodeClass parentClass, Langua
             baseSuffix = " : base()";
         var parameters = string.Join(", ", code.Parameters.OrderBy(x => x, parameterOrderComparer).Select(p => conventions.GetParameterSignature(p, code)).ToList());
         var methodName = isConstructor ? parentClass.Name.ToFirstCharacterUpperCase() : code.Name.ToFirstCharacterUpperCase();
-        var includeNullableReferenceType = code.Parameters.Any(static codeParameter => codeParameter.IsOfKind(CodeParameterKind.RequestConfiguration));
+        var includeNullableReferenceType = code.Parameters.Any(static codeParameter => codeParameter.IsOfKind(CodeParameterKind.RequestConfiguration) || codeParameter.Optional || codeParameter.Type.IsNullable);
```

Comment:
I think we need to double check this as it leads to build issues when,

- An optional parameter like `CancellationToken`  is set as a nullable parameter. There really is no need for this as the default value will `CancellationToken.None`

Reply:
I've removed the test for optional parameters.